.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,258 @@
+name: Build LLDAP Bin and Dockerize
+
+on:
+  push:
+    branches: [ '*' ] #rolling build on main only
+    tags: 'v*' #tag for specific release v1.0.2,or anytag, start with v
+  pull_request:
+    branches: [ 'main' ] #build test on main, not pushing
+  workflow_dispatch:
+
+env:
+  CARGO_TERM_COLOR: always
+  
+jobs:
```

Comment:
Can you add some comments about how sections are organized and what they do? It's not self-evident.

Reply:
Added some comments for more description